docs/source-literals.md

```diff
@@ -0,0 +1,186 @@
+    function RunFromZombie()
+        print FUNCTION_NAME
+    end function
+end namespace
+```
+
+transpiles to: 
+
+```BrightScript
+function Human_Versus_Zombie_RunFromZombie()
+    print "Human_Versus_Zombie_RunFromZombie"
+end function
+```
+### Anonymous Functions
+For anonymous functions, FUNCTION_NAME will include the enclosing function name, followed by `anon` and the index of the function in its parent. If there are multiple nested anonymous functions, each level will have its own `anon[index]` in the name. Function indexes start at 0 within each enclosing function.
+
+```BrighterScript
+function main()
+    zombieAttack = function()
+        print FUNCTION_NAME
+    end function
+
+    humanAttack = function()
+        useSword = function()
+            print FUNCTION_NAME
+        end function
+        useSword()
+    end function
+end function
+namespace Human.Versus.Zombie
+    function Eat()
+        print SOURCE_FUNCTION_NAME
+        findFood = function()
+            print FUNCTION_NAME
+        end function
+    end function
+end namespace
+```
+
+transpiles to: 
+
+```BrightScript
+function main()
+    zombieAttack = function()
+        print "main$anon0"
+    end function
+    humanAttack = function()
+        useSword = function()
+            print "main$anon1$anon0"
+        end function
+        useSword()
+    end function
+end function
+function Human_Versus_Zombie_Eat()
+    print "Human.Versus.Zombie.Eat"
+    findFood = function()
+        print "Human_Versus_Zombie_Eat$anon0"
+    end function
+end function
+```
+
+## SOURCE_FUNCTION_NAME
+This works the same as FUNCTION_NAME except that it will use a dot in namespace names instead of the transpiled underscores.
+
+
+```BrighterScript
+namespace Human.Versus.Zombie
+    function Eat()
+        print SOURCE_FUNCTION_NAME
+        findFood = function()
+            print SOURCE_FUNCTION_NAME
+        end function
+    end function
+end namespace
+```
+
+transpiles to: 
+
+```BrightScript
+function Human_Versus_Zombie_Eat()
+    print "Human.Versus.Zombie.Eat"
+    findFood = function()
+        print "Human.Versus.Zombie.Eat$anon0"
+    end function
+end function
+```
+
+## SOURCE_LOCATION
+A combination of SOURCE_FILE_PATH and SOURCE_LINE_NUM.
+
+```BrighterScript
+function main()
+    print SOURCE_LOCATION
+end function
+```
+
+transpiles to: 
+
+```BrightScript
+function main()
+    print "file:/c:\projects\roku\brighterscript\scripts\rootDir\source\main.bs:2"
+end function
+```
+
+## PKG_PATH
+The pkg path of the file.
+
+```BrighterScript
+function main()
+    print PKG_PATH
+end function
+```
+
+transpiles to: 
+
+```BrightScript
+function main()
+    print "pkg:/source/main.brs"
+end function
+```
+
+## PKG_LOCATION
+A combination of PKG_PATH and LINE_NUM. Keep in mind, LINE_NUM is a runtime variable provided by Roku. 
+```BrighterScript
+print PKG_LOCATION
+```
+
+transpiles to: 
+
+```BrightScript
+print "pkg:/source/main.brs:" + str(LINE_NUM)
```

Comment:
Was there a reason for using Roku’s vs the SOURCE_LINE_NUM?

Reply:
I figure since the pkgPath is basically the path to the file at runtime, it made sense to use the runtime line number too.

Reply:
@triwav do you agree with that approach, or do you think this should be something else?